src/externalUser.ts

```diff
@@ -0,0 +1,81 @@
+import { KeyManager, KeyManagerLevel } from './services/keymanager';
+import { JWTLoginPayload, OnPressLoginOptions } from './userApps';
+import { generateRandomKeyPair, randomString } from './util/crypto';
+import { ES256KSigner } from '@tonomy/did-jwt';
+import { createJWK, toDid } from './util/did-jwk';
+import { Message } from './util/message';
+import { getSettings } from './settings';
+
+export class ExternalUser {
+    static async loginWithTonomy(
+        { redirect = true, callbackPath }: OnPressLoginOptions,
+        keyManager: KeyManager,
+        keyManagerLevel: KeyManagerLevel = KeyManagerLevel.BROWSER_LOCAL_STORAGE
+    ): Promise<string | void> {
+        const { privateKey, publicKey } = generateRandomKeyPair();
+
+        if (keyManager) {
+            await keyManager.storeKey({
+                level: keyManagerLevel,
+                privateKey: privateKey,
+            });
+        }
+
+        const payload: JWTLoginPayload = {
+            randomString: randomString(32),
+            origin: window.location.origin,
+            publicKey: publicKey.toString(),
+            callbackPath,
+        };
+
+        // TODO use expiresIn to make JWT expire after 5 minutes
+
+        const signer = ES256KSigner(privateKey.data.array, true);
+        const jwk = await createJWK(publicKey);
+
+        const issuer = toDid(jwk);
+
+        const token = (await Message.sign(payload, { did: issuer, signer: signer as any, alg: 'ES256K-R' })).jwt;
```

Comment:
if we want to move the external logic to here shouldn't we also move the sign Message and verify ?

Reply:
yes maybe, but in a future PR